Deleting enrollment

CelsoDeCarvalho opened the pull request:
- Enable delete enrollment
- Enhace hader filters

CelsoDeCarvalho committed:
feat: add function to delete enrollment

CelsoDeCarvalho committed:
enhace: add load on enrollment deletion and apply header filters to table

cnandza commented on src/pages/enrollment/enrollment.tsx:
```diff
@@ -8,19 +8,21 @@ import ModalManager from "../../components/modal/ModalManager";
+    const [filterState, setFilterState] = useState<{ dataElements: any, attributes: any }>({ attributes: [], dataElements: [] });
     const [refetch,] = useRecoilState(TableDataRefetch);
 
     const handleOpenEditModal = (e: Record<string, any>) => {
```
why we don´t simplify thiiis ?

const handleOpenModal = (type: "edit" | "delete", e?: Record<string, any>) => {
    if (e) {
        add("trackedEntity", e?.row?.trackedEntity);
        add("enrollment", e?.row?.enrollmentId);
    }

    if (type === "edit") setOpenEditModal(true);
    else setOpenDeleteModal(true);
};

useEffect(() => {
    if (!openEditModal && !openDeleteModal) {
        remove("trackedEntity");
        remove("enrollment");
    }
}, [openEditModal, openDeleteModal]);

cnandza commented on src/components/modal/deleteEnrollment/ModalContent.tsx:
```diff
@@ -0,0 +1,100 @@
+                        destructive={true}
+                        submitButtonLabel={loading ? "Deleting" : "Delete"}
+                        onFormSubtmit={(e) => { onSubmit(e) }}
+                        onInputChange={(e) => { onChange(e) }}
```
why you have onChange here?

cnandza commented on src/components/modal/deleteEnrollment/ModalManager.tsx:
```diff
@@ -0,0 +1,73 @@
+import { useRecoilState } from 'recoil';
+
+const ModalManagerEnrollmentDelete = (props: ModalManagerInterface) => {
+    const { loadingCodes, generatedVariables } = useGetPatternCode();
```
we don´t need this here

cnandza commented on src/components/modal/deleteEnrollment/ModalManager.tsx:
```diff
@@ -0,0 +1,73 @@
+    const programsValues = useProgramsKeys();
+    const programData = programsValues[0];
+    const { open, setOpen } = props;
+    const dataStoreData = useDataStoreKey({ sectionType: "student" });
```
don´t hardcode sectionType

cnandza commented on src/components/modal/deleteEnrollment/ModalManager.tsx:
Remove the unused functions imported here

cnandza commented on src/hooks/enrollment/useDeleteEnrollment.ts:
We can put this in the functions lib... Its specif for semis enrollment, but we can need later

cnandza commented on src/hooks/enrollment/useGetEnrollment.ts:
We can put this in the functions lib... Its specif for semis, but we can need later

CelsoDeCarvalho committed:
enhance(useGetDeleteEnrollmentInitialValues.ts): removed unecessary fetch

CelsoDeCarvalho committed:
refactor: remove unecessary code

CelsoDeCarvalho committed:
feat:  enrollment delete strategy

If tei has one enrollment, then delete the tei, if it has more than one, delete the selected enrollment
